src/olympia/reviewers/views.py

```diff
@@ -439,6 +439,19 @@ def determine_channel(channel_as_text):
     return channel, content_review
 
 
+def other_queue_has_unreviewed_versions(addon, channel):
+    """Return True if the addon has unreviewed versions in another queue."""
+    versions = addon.versions.filter(
+        channel=amo.CHANNEL_LISTED
+        if channel == amo.CHANNEL_UNLISTED
+        else amo.CHANNEL_UNLISTED,
+    )
+    for version in versions:
+        if version.is_unreviewed:
+            return True
+    return False
```

Comment:
File reviewed status isn't really the same thing as having a due_date - there can be unreviewed versions that don't need attention (e.g. because the auto approve task is just about to pick them up); and also versions with a reviewed status that have been flagged for needs human review.

Filter on `due_date` instead.

Reply:
Also, doing a loop on `versions` like that, you would be making a database query against all versions for the add-on, loading them all, then building a `Version` instance for each loop iteration, just to check that there was something.

Filter on `due_date__isnull=False`, and use `.exists()` on that queryset, no need to loop.

Finally, use `addon.versions(manager='unfiltered_for_relations')`, otherwise you won't see deleted versions in your queryset, and you want to have them in case we need to review something the developer deleted.